### DISCUSSION TITLE
Could pinia work on backend,How to do it without vue component?

### DISCUSSION THREAD

**Discussion by radiorz** [vuejs/pinia#1050] [Help and Questions] [answered]
Could pinia work without vue component?

**Comment by cefn** [comment, marked as answer]
Yes, Pinia can define state and business logic without being imported into a Vue component.

I am currently using it this way, following the setup function style of `defineStore`, basically defining state mechanisms just like in a Vue component's `setup` call, but with no template or rendering.

**Reply by radiorz** [reply]
Thank you for your answer. I would appreciate it if there was an example

**Reply by cefn** [reply]
Is this example any good https://github.com/vuejs/pinia/issues/1061#issuecomment-1040586859 it's a simple Jest test that authors some state types, creates some store definitions and then creates stores from them. It then runs a test to prove that you can run an action and make something happen. There are no Vue components involved.
